server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/ClusterBalanceStatsTests.java

```diff
@@ -260,15 +281,371 @@ public void testStatsForNoIndicesInTier() {
                         )
                     ),
                     Map.ofEntries(
-                        Map.entry("node-1", new NodeBalanceStats("node-1", List.of(DATA_CONTENT_NODE_ROLE.roleName()), 0, 0, 0.0, 0L, 0L)),
-                        Map.entry("node-2", new NodeBalanceStats("node-2", List.of(DATA_CONTENT_NODE_ROLE.roleName()), 0, 0, 0.0, 0L, 0L)),
-                        Map.entry("node-3", new NodeBalanceStats("node-3", List.of(DATA_CONTENT_NODE_ROLE.roleName()), 0, 0, 0.0, 0L, 0L))
+                        Map.entry(
+                            "node-1",
+                            new NodeBalanceStats("node-1", List.of(DATA_CONTENT_NODE_ROLE.roleName()), 0, 0, 0.0, 0L, 0L, null)
+                        ),
+                        Map.entry(
+                            "node-2",
+                            new NodeBalanceStats("node-2", List.of(DATA_CONTENT_NODE_ROLE.roleName()), 0, 0, 0.0, 0L, 0L, null)
+                        ),
+                        Map.entry(
+                            "node-3",
+                            new NodeBalanceStats("node-3", List.of(DATA_CONTENT_NODE_ROLE.roleName()), 0, 0, 0.0, 0L, 0L, null)
+                        )
                     )
                 )
             )
         );
     }
 
+    // NodeBalanceStats specific tests //
+
+    private static final String UNKNOWN = "UNKNOWN";
+    private static final int UNDESIRED_SHARD_ALLOCATION_DEFAULT_VALUE = -1;
+
+    public void testReadFrom() throws IOException {
```

Comment:
I appreciate the effort on writing these wire serialization tests manually. But I think it's better to have these in a separate test class that extends `AbstractWireSerializingTestCase`.

We also have `AbstractXContentSerializingTestCase` for xcontent serialization. But since we don't normally parse from xcontent these days, it is also common to not relying on the abstract class and just test the `toXContent` manually. I think they can be added in the same test class for wire serialization.